Contentful.Core/ContentfulManagementClient.cs

```diff
@@ -513,6 +557,64 @@ public ContentfulManagementClient(HttpClient httpClient, IOptions<ContentfulOpti
+        /// <param name="contentTypeId">The id of the <see cref="ContentType"/> of the entry. Need only be set if you are creating a new entry.</param>
+        /// <param name="version">The last known version of the entry. Must be set when updating an entry.</param>
+        /// <param name="cancellationToken">The optional cancellation token to cancel the operation.</param>
+        /// <returns>The created or updated entry.</returns>
+        public async Task<T> CreateOrUpdateEntryAsync<T>(T entry, string id, string spaceId = null, string contentTypeId = null, int? version = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var entryToCreate = new Entry<dynamic>();
+            entryToCreate.SystemProperties = new SystemProperties
+            {
+                Id = id
+            };
+            entryToCreate.Fields = entry;
+
+            var createdEntry = await CreateOrUpdateEntryAsync(entryToCreate, spaceId, contentTypeId, version, cancellationToken);
+            return (createdEntry.Fields as JObject).ToObject<T>();
+        }
+
+
+        /// <summary>
+        /// Updates an entry fields for a certain locale using the values from the provided object.
+        /// </summary>
+        /// <param name="entry">The object to use as values for the entry fields.</param>
+        /// <param name="id">The id of the entry to update.</param>
+        /// <param name="locale">The locale to set the fields for. The default locale for the space will be used if this parameter is null.</param>
+        /// <param name="spaceId">The id of the space. Will default to the one set when creating the client.</param>
+        /// <param name="cancellationToken">The optional cancellation token to cancel the operation.</param>
+        /// <returns>The created or updated <see cref="Entry{T}"/>.</returns>
+        public async Task<Entry<dynamic>> UpdateEntryForLocale(object entry, string id, string locale = null, string spaceId = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var entryToUpdate = await GetEntryAsync(id, spaceId);
+            
+            if(locale == null)
+            {
+                locale = (await GetLocalesCollectionAsync(spaceId, cancellationToken)).FirstOrDefault(c => c.Default).Code;
+            }
+
+            var jsonEntry = JObject.Parse(ConvertObjectToJsonString(entry));
+            var fieldsToUpdate = (entryToUpdate.Fields as JObject);
+
+            foreach (var prop in fieldsToUpdate.Children().Where(p => p is JProperty).Cast<JProperty>())
+            {
+                if(jsonEntry[prop.Name] != null)
+                {
+                    fieldsToUpdate[prop.Name][locale] = jsonEntry[prop.Name];
+                }
+            }
+
+            var updatedEntry = await CreateOrUpdateEntryAsync(entryToUpdate,spaceId: spaceId, version: entryToUpdate.SystemProperties.Version, cancellationToken: cancellationToken);
```

Comment:
Excuse my C#, since `UpdateEntryFroLocale` is aync, is there a reason to `await` the `CreateOrUpdateEntryAsync` inside of the method and return the updatedEntry ?